chaindataloader/binary_test.go

```diff
@@ -0,0 +1,180 @@
+package chaindataloader
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightninglabs/neutrino/headerfs"
+)
+
+const (
+	testDataDir = "testdata/"
+)
+
+// TestNewBinarySideLoad tests that the reader returns the appropriate data
+// required for fetching headers when initialized.
+func TestNewBinarySideLoad(t *testing.T) {
+	type testSideLoadConfig struct {
+		name        string
+		path        string
+		endHeight   int64
+		startHeight int64
+		network     wire.BitcoinNet
+		tipHeight   int64
+		err         bool
```

Comment:
Rather than asserting if an error is present or not, it should assert the error directly. So changing this to be `error`, same below.

Reply:
This makes the test more precise, as we make sure we're getting the exact error we expect.

Reply:
Doing that makes the test to always fail even with the same error message.

Reply:
> Doing that makes the test to always fail even with the same error message.

Have an example? You can use `error.Is` an `errors.As` to ensure the the `error` is what you expect.